src/Microsoft.Extensions.DependencyInjection.Abstractions/IDesignTimeServiceProviderFactory.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// A factory for creating the application's <see cref="IServiceProvider"/> at design time. Implement this interface
+    /// to enable design-time services that need access to the application's services. Design-time services will
+    /// automatically discover implementations of this interface.
+    /// </summary>
+    public interface IDesignTimeServiceProviderFactory
```

Comment:
Interfaces have the semantics we assign to them :smile_cat: What makes `IServiceScope` something more than a way to get a service provider? Is it really just the fact that it implements `IDisposable` or does the name and our shared knowledge of what it is for that make it special?

I am fine with passing the args in the constructor unless there are technical issues with it. I actually had the same thoughts a few days ago and I already forgot why I abandoned them :smile: